cli_tools/diagnostics/main_windows_test.go

```diff
@@ -0,0 +1,116 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// These test can only be run on windows, as the functions are highly dependent on windows OS.
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+const (
+	pathNotExistErr    = "The system cannot find the path specified."
+	fileNotExistErr    = "The system cannot find the file specified."
+	systemLogPath      = `C:\Windows\System32\winevt\Logs\System.evtx`
+	kubeletLogFileName = "kubelet.log"
+)
+
+func pathNonExist(e error) bool {
+	if strings.Contains(e.Error(), pathNotExistErr) {
+		return true
+	}
+	return false
+}
+
+func fileNonExist(e error) bool {
+	if strings.Contains(e.Error(), fileNotExistErr) {
+		return true
+	}
+	return false
+}
+
+func TestCollectFilePaths(t *testing.T) {
+	dir := os.TempDir()
+	testRoot := filepath.Join(dir, "collectFilePathsTest")
+	defer os.RemoveAll(testRoot)
+	kubeletlogFilePath := filepath.Join(testRoot, kubeletLogFileName)
+	os.Create(kubeletlogFilePath)
+	nonExistFilePath := filepath.Join(dir, "xxx")
+	type args struct {
+		roots []string
+	}
+	tests := []struct {
+		name  string
+		args  args
+		want  []string
+		errOK func(error) bool
+	}{
+		{"Nil roots", args{nil}, []string{}, nil},
+		{"Empty roots", args{[]string{""}}, []string{}, pathNonExist},
+		{"Existing roots", args{[]string{testRoot}}, []string{kubeletlogFilePath}, nil},
+		{"Non-existing paths", args{[]string{nonExistFilePath}}, []string{}, fileNonExist},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotFiles, gotErrs := collectFilePaths(tt.args.roots)
+			if !reflect.DeepEqual(gotFiles, tt.want) {
+				t.Errorf("unexpected filepaths, want %v, got %v", tt.want, gotFiles)
+			}
+			for _, err := range gotErrs {
+				if tt.errOK == nil || !tt.errOK(err) {
+					t.Errorf("collectFilePaths() got unexpected error = %v", gotErrs)
+				}
+			}
+		})
+	}
+}
+
+func include(ss []string, str string) bool {
```

Comment:
Either add a comment describing what this function does, or name it more specifically: stringArrayIncludesString ?

ss and str are also not great argument names; maybe "string_array" or "search_array", and "target"?

Reply:
Agree. And the golang suggest mixedCap naming convention, so change to stringArray. https://golang.org/doc/effective_go.html

Reply:
Nice, looks much better!